### PR TITLE
Add float16 docs

### PR DESCRIPTION
**Pull request by TomWildenhain-Microsoft**
Docs for float16.py from onnxconverter_common

Preview here:
https://tomwildenhain-microsoft.github.io/onnxruntime/docs/how-to/float16.html

Updated preview here: https://natke.github.io/onnxruntime/docs/performance/float16.html

### CODE REVIEW

File: docs/how-to/float16.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+If float16 conversion is giving poor results, you can convert most of the ops to float16 but leave some in float32. The `auto_mixed_precision` tool finds a minimal set of ops to skip while retaining a certain level of accuracy. You will need to provide a sample input for the model. 
+
+Since the CPU version of ONNX Runtime doesn't support float16 ops and the tool needs to measure the accuracy loss, **the mixed precision tool must be run on a device with a GPU**.
+
+```python
+from onnxconverter_common import auto_mixed_precision
+import onnx
+
+model = onnx.load("path/to/model.onnx")
+model_fp16 = auto_convert_mixed_precision(model, test_data, rtol=0.01, atol=0.001, keep_io_types=True)
```
[the remark this answers is not in the document]
Comment (natke) [reply docs/how-to/float16.md]: @TomWildenhain-Microsoft Can you check Ed's comment please?

[the remark this answers is not in the document]
Comment (TomWildenhain-Microsoft) [reply docs/how-to/float16.md]: Yes, @edgchen1 I think you are right.

[the remark this answers is not in the document]
Comment (snnn) [reply docs/how-to/float16.md]: @natke , is this PR still needed?

[the remark this answers is not in the document]
Comment (natke) [reply docs/how-to/float16.md]: Yes, I'm looking for someone to look over it in Tom's absence. Any ideas? @snnn

**Pull request merged**